### ISSUE TITLE
Can ArchUnit check that assertions include a message?

### ISSUE THREAD

**Issue by roxspring** [TNG/ArchUnit#963]
I ran into a situation today where our production code hit an assertion without a message, resulting in an `AssertionError` being thrown e.g.:

```java
String text = "Hello";
assert text.isEmpty();
```

Clearly this would have been easier to diagnose if the assertion had a n associated message e.g.:

```java
String text = "Hello";
assert text.isEmpty() : "Text should have been empty: " + text;
```

A couple of us immediately thought "We should create an ArchUnit test to require that all assertions statements include a message", but I can't see how this would be expressed using ArchUnit. 

Is this even possible in ArchUnit?

Or is this not really considered architectural and I should be looking somewhere like SpotBugs instead?

**Comment by roxspring** [comment]
2h later and I eventually realise that `assert condition` in java essentially compiles down to `if (!condition) throw new AssertionError()`, and therefore looking for calls to the `AssertionError` no-arg constructor is a pretty close approximation:

```java
    ArchRuleDefinition.noClasses()
      .should()
      .callConstructor(AssertionError.class)
      .as("assertions should have a message")
      .check(classes);
```

**Comment by hankem** [comment]
Should we include this example in [com.tngtech.archunit.library.GeneralCodingRules](https://github.com/TNG/ArchUnit/blob/main/archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java)?

**Comment by hankem** [comment]
FYI @roxspring : This is now available in [ArchUnit v1.1.0](https://github.com/TNG/ArchUnit/releases/tag/v1.1.0).
